test/cql-pytest/test_secondary_index.py

```diff
@@ -185,3 +186,19 @@ def test_paging_with_desc_clustering_order(cql, test_keyspace):
             cql.execute(f"INSERT INTO {table}(p,c) VALUES ({i}, 42)")
         stmt = SimpleStatement(f"SELECT * FROM {table} WHERE c = 42", fetch_size=1)
         assert len([row for row in cql.execute(stmt)]) == 3
+
+# Selecting values using only clustering key should require filtering, but work correctly
+# Reproduces issue #8991
+@pytest.mark.xfail(reason="Select from indexed table using only clustering key breaks. Issue #8991")
+def test_filter_cluster_key(cql, test_keyspace):
+    schema = 'p int, c1 int, c2 int, primary key (p, c1, c2)'
+    extra = ''
+    with new_test_table(cql, test_keyspace, schema, extra) as table:
```

Comment:
extra='' is the default, you don't need to pass it to new_test_table.

Reply:
done